### PR TITLE
[TEST] Should create an order with product with fixed promotion

### PR DESCRIPTION
**Pull request by yellowee**
I want to merge this change because it adds test for creating an order with product with fixed promotion CORE_2101

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

File: saleor/tests/e2e/orders/test_order_products_on_fixed_promotion.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    input = {"shippingMethod": shipping_method_id}
+    draft_update = draft_order_update(e2e_staff_api_client, order_id, input)
+    order_shipping_id = draft_update["order"]["deliveryMethod"]["id"]
+    shipping_price = draft_update["order"]["shippingPrice"]["gross"]["amount"]
+    subtotal_gross_amount = draft_update["order"]["subtotal"]["gross"]["amount"]
+    total_gross_amount = draft_update["order"]["total"]["gross"]["amount"]
+    assert order_shipping_method_id == order_shipping_id
+    print(draft_update)
+
+    # Step 4 - Complete the draft order
```
Comment (szczecha) [inline saleor/tests/e2e/orders/test_order_products_on_fixed_promotion.py:220]: Can you add an assertion that data.draftOrderComplete.order.lines[0].unitDiscountReason == Promotion rules discounts: {promotion_name}: {promotion_rule_name}

**Review by szczecha — CHANGES_REQUESTED** [review]

File: saleor/tests/e2e/orders/__init__.py
[the remark this answers is not in the document]
Comment (yellowee) [reply saleor/tests/e2e/orders/__init__.py:1]: https://github.com/saleor/saleor/pull/13639/commits/91a63d3c240131458d572a698fe6cac4b3077a58

File: saleor/tests/e2e/orders/test_order_products_on_fixed_promotion.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    )
+
+    # Step 1 - Create a draft order for a product with fixed promotion
+    lines = [{"variantId": product_variant_id, "quantity": 1}]
+    input = {
+        "channelId": channel_id,
+        "billingAddress": DEFAULT_ADDRESS,
+        "shippingAddress": DEFAULT_ADDRESS,
+        "shippingMethod": shipping_method_id,
+        "lines": lines,
```
[the remark this answers is not in the document]
Comment (yellowee) [reply saleor/tests/e2e/orders/test_order_products_on_fixed_promotion.py]: https://github.com/saleor/saleor/pull/13639/commits/91a63d3c240131458d572a698fe6cac4b3077a58

File: saleor/tests/e2e/orders/test_order_products_on_fixed_promotion.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+    assert data["order"]["billingAddress"] is not None
+    assert data["order"]["shippingAddress"] is not None
+    order_shipping_method_id = data["order"]["shippingMethods"][0]["id"]
+    unit_price = float(variant_price) - float(discount_value)
+    undiscounted_price = data["order"]["lines"][0]["undiscountedUnitPrice"]["gross"][
+        "amount"
+    ]
+    assert float(undiscounted_price) == float(variant_price)
+    assert data["order"]["lines"][0]["unitPrice"]["gross"]["amount"] == unit_price
+    print(data)
```
[the remark this answers is not in the document]
Comment (yellowee) [reply saleor/tests/e2e/orders/test_order_products_on_fixed_promotion.py]: https://github.com/saleor/saleor/pull/13639/commits/91a63d3c240131458d572a698fe6cac4b3077a58

File: saleor/tests/e2e/orders/test_order_products_on_fixed_promotion.py [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+    # UNCOMMENT WHEN https://github.com/saleor/saleor/issues/13675 is resolved:
+    # assert order_shipping_method_id == shipping_method_id
+    assert order_id is not None
+
+    # Step 2 - Add order lines to the order
+    lines = [{"variantId": product_variant_id, "quantity": 1}]
+    order_lines = order_lines_create(e2e_staff_api_client, order_id, lines)
+    order_product_variant_id = order_lines["order"]["lines"][0]["variant"]["id"]
+    assert order_product_variant_id == product_variant_id
+    print(order_lines)
```
[the remark this answers is not in the document]
Comment (yellowee) [reply saleor/tests/e2e/orders/test_order_products_on_fixed_promotion.py]: https://github.com/saleor/saleor/pull/13639/commits/91a63d3c240131458d572a698fe6cac4b3077a58

File: saleor/tests/e2e/orders/test_order_products_on_fixed_promotion.py (hunk 1 above)
Comment (yellowee) [reply saleor/tests/e2e/orders/test_order_products_on_fixed_promotion.py:219]: https://github.com/saleor/saleor/pull/13639/commits/91a63d3c240131458d572a698fe6cac4b3077a58

File: saleor/tests/e2e/orders/utils/draft_order_update.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        variables=variables,
+    )
+    content = get_graphql_content(response)
+
+    data = content["data"]["draftOrderUpdate"]
+
+    order_id = data["order"]["id"]
+    # errors = data["errors"]
+
+    # assert errors == []
```
[the remark this answers is not in the document]
Comment (yellowee) [reply saleor/tests/e2e/orders/utils/draft_order_update.py]: https://github.com/saleor/saleor/pull/13639/commits/91a63d3c240131458d572a698fe6cac4b3077a58

**Review by szczecha — COMMENTED** [review]
Looks better now 👍 One last thing please fix the formatting in all added/changed mutations there is some mess.

**Review by szczecha — APPROVED** [review]

**Comment by IKarbowiak** [comment]
We should merge only when the PR has at least two approves.

**Pull request merged**